exec/java-exec/src/main/java/org/apache/drill/exec/expr/ClassGenerator.java

```diff
@@ -86,14 +88,16 @@ public static MappingSet getDefaultMapping() {
   }
 
   @SuppressWarnings("unchecked")
-  ClassGenerator(CodeGenerator<T> codeGenerator, MappingSet mappingSet, SignatureHolder signature, EvaluationVisitor eval, JDefinedClass clazz, JCodeModel model) throws JClassAlreadyExistsException {
+  ClassGenerator(CodeGenerator<T> codeGenerator, MappingSet mappingSet, SignatureHolder signature, EvaluationVisitor eval, JDefinedClass clazz, JCodeModel model, OptionManager optionManager) throws JClassAlreadyExistsException {
     this.codeGenerator = codeGenerator;
     this.clazz = clazz;
     this.mappings = mappingSet;
     this.sig = signature;
     this.evaluationVisitor = eval;
     this.model = model;
-    blocks = (LinkedList<JBlock>[]) new LinkedList[sig.size()];
+    this.optionManager = optionManager;
```

Comment:
Looks like the `optionManager` being passed down is system level. Make this explicit by renaming `optionManager` to `systemOptions`.

Users also need to be aware.